### PR TITLE
feat: load channels from backend

### PR DESCRIPTION
**Pull request by erikarenhill**
Previously the appstore frontend had a static config for release channels, now they are being loaded dynamically from the backend instead. This PR also contains a few bug fixes regarding channels when editing/creating versions and uploading new apps.

### CODE REVIEW

File: client/src/components/appVersion/VersionListEdit.jsx [hunk 1]
```diff
@@ -185,11 +187,22 @@ class VersionListEdit extends Component {
             </IconButton>
         )
 
+        const DHISReleaseChannels = this.props.channels.channels.map(
+            (channel, i) => (
```
Comment (HendrikThePendric) [inline client/src/components/appVersion/VersionListEdit.jsx]: `i` is unused

File: client/src/components/form/UploadAppFormStepper.jsx [hunk 2]
```diff
@@ -132,6 +133,13 @@ AppGeneralSection.defaultProps = {
 }
 
 const AppVersionSection = props => {
+    const releaseChannels = props.channels.map((channel, i) => (
```
Comment (HendrikThePendric) [inline client/src/components/form/UploadAppFormStepper.jsx]: `i` is unused

File: client/src/components/appVersion/VersionListEdit.jsx [hunk 3]
```diff
@@ -139,7 +141,7 @@ class VersionListEdit extends Component {
         this.setState({
             editedValues: merge(editedValues, {
                 [versionId]: {
-                    channel: config.ui.releaseChannels[selectedIndex],
+                    channel: this.props.channels.channels[selectedIndex].name,
```
[the remark this answers is not in the document]
Comment (erikarenhill) [reply client/src/components/appVersion/VersionListEdit.jsx]: @HendrikThePendric well I agree the naming could be confusing, but the channels in redux also contain flags such as loading, loaded, while the channels one deeper is the actual list of objects/data. Maybe we could just change the name of the second one to "list" ?

[the remark this answers is not in the document]
Comment (erikarenhill) [reply client/src/components/appVersion/VersionListEdit.jsx]: the selectedIndex is coming from a dropdown component using the same array as source so it should not be able to be undefined

File: client/src/components/appVersion/VersionListEdit.jsx (hunk 1 above)
Comment (erikarenhill) [reply client/src/components/appVersion/VersionListEdit.jsx]: removed it

File: client/src/components/form/UploadAppFormStepper.jsx (hunk 2 above)
Comment (erikarenhill) [reply client/src/components/form/UploadAppFormStepper.jsx]: removed it

File: client/src/components/appVersion/VersionListEdit.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (HendrikThePendric) [reply client/src/components/appVersion/VersionListEdit.jsx]: > Maybe we could just change the name of the second one to "list" ?

Yeah IMO `list` or `collection` would make things a lot clearer.

[the remark this answers is not in the document]
Comment (erikarenhill) [reply client/src/components/appVersion/VersionListEdit.jsx]: changed to list

**Pull request merged**